lib/enkf/enkf_main_jobs.cpp

```diff
@@ -505,10 +507,14 @@ void * enkf_main_std_scale_correlated_obs_JOB(void * self, const stringlist_type
     int ensemble_size                       = enkf_main_get_ensemble_size(enkf_main);
     enkf_fs_type * fs                       = enkf_main_job_get_fs( enkf_main );
     enkf_obs_type * obs                     = enkf_main_get_obs( enkf_main );
-    int_vector_type * realizations          = int_vector_alloc(1, 0);
     local_obsdata_type * obsdata = local_obsdata_alloc( "OBS-JOB" );
 
-    int_vector_init_range(realizations, 0, ensemble_size, 1);
+    // Find out what realizations have data
+    auto const* state_map = enkf_fs_get_state_map(fs);
+    assert(ensemble_size == state_map_get_size(state_map));
```

Comment:
This `assert` is not required; we will just infer the interesting/relevant size from the `state_map`.